app-libs/stf/src/stf_sgx.rs

```diff
@@ -182,6 +182,14 @@ impl Stf {
 					Self::shield_funds(who, value)?;
 					Ok(())
 				},
+				TrustedCall::new_game(sender, opponent) => {
+					let origin = sgx_runtime::Origin::signed(sender.clone());
+					debug!("collect four new_game ({:x?}, {:x?})", sender.encode(), opponent.encode());
+					sgx_runtime::ConnectfourCall::<Runtime>::new_game {	player_one: opponent.clone(), player_two: opponent.clone() }
```

Comment:
That's twice the same player.

Also, we don't have a sender here, as the node matches the players. I'd suggest to rename to player_one or player_red & blue instead of sender and opponent, which might lead to confusion